Implement fail-fast for TestInfrastructure_GenerateResources

## Problem

When `TestInfrastructure_GenerateResources` fails, subsequent infrastructure verification tests still run and fail with confusing errors, since they depend on the generated files (`credentials.yaml`, `is.yaml`, `aro.yaml`).

Example failure:
```
--- FAIL: TestInfrastructure_GenerateResources (1.21s)
```

After this failure, the following tests will fail unnecessarily:
- `TestInfrastructure_VerifyCredentialsYAML`
- `TestInfrastructure_VerifyInfrastructureSecretsYAML`
- `TestInfrastructure_VerifyAROClusterYAML`

## Proposed Solution

Implement a fail-fast mechanism so that when `TestInfrastructure_GenerateResources` fails, dependent tests are skipped with a clear message.

### Implementation approach

Use a shared state variable (similar to the pattern in other test phases) to track if generation succeeded:

```go
var infrastructureGenerationSucceeded bool

func TestInfrastructure_GenerateResources(t *testing.T) {
    // ... existing logic ...
    
    // On success, set the flag
    infrastructureGenerationSucceeded = true
}

func TestInfrastructure_VerifyCredentialsYAML(t *testing.T) {
    if !infrastructureGenerationSucceeded {
        t.Skip("Skipping: TestInfrastructure_GenerateResources did not succeed")
    }
    // ... rest of test ...
}
```

### Test flow with fail-fast

```
│  Test 1: GenerateResources                                       │
│  ├── Set environment variables                                   │
│  ├── cd to ARO_REPO_DIR                                          │
│  ├── Run: bash aro-hcp-gen.sh <output-dir>                       │
│  └── Verify output files created                                 │
│      └── On FAIL → set flag, subsequent tests skip               │
│                                                                   │
│  Test 2: VerifyCredentialsYAML                                   │
│  ├── Check fail-fast flag → SKIP if generation failed            │
│  └── Validate credentials.yaml                                   │
│                                                                   │
│  Test 3: VerifyInfrastructureSecretsYAML                         │
│  ├── Check fail-fast flag → SKIP if generation failed            │
│  └── Validate is.yaml                                            │
│                                                                   │
│  Test 4: VerifyAROClusterYAML                                    │
│  ├── Check fail-fast flag → SKIP if generation failed            │
│  └── Validate aro.yaml                                           │
```

## Acceptance Criteria

- [ ] When `TestInfrastructure_GenerateResources` fails, subsequent verification tests are skipped
- [ ] Skip message clearly indicates the reason (generation failed)
- [ ] When generation succeeds, all verification tests run normally
- [ ] Pattern is consistent with fail-fast in other test phases

## Files to modify

- `test/04_generate_yamls_test.go`

## Related

This follows the same pattern that should be applied to other phases with dependent tests.

--- a/test/04_generate_yamls_test.go
+++ b/test/04_generate_yamls_test.go
@@ -6,6 +6,10 @@ import (
 	"testing"
 )
 
+// infrastructureGenerationSucceeded tracks whether TestInfrastructure_GenerateResources completed successfully.
+// When false, dependent verification tests will skip to avoid confusing cascading failures.
+var infrastructureGenerationSucceeded bool
+
 // TestInfrastructure_GenerateResources tests generating ARO infrastructure resources
 func TestInfrastructure_GenerateResources(t *testing.T) {
 
@@ -104,10 +108,19 @@ func TestInfrastructure_GenerateResources(t *testing.T) {
 		}
 	}
 	PrintToTTY("\n")
+
+	// Mark generation as successful only if no errors occurred
+	if !t.Failed() {
+		infrastructureGenerationSucceeded = true
+	}
 }
 
 // TestInfrastructure_VerifyCredentialsYAML verifies credentials.yaml exists and is valid
 func TestInfrastructure_VerifyCredentialsYAML(t *testing.T) {
+	if !infrastructureGenerationSucceeded {
+		t.Skip("Skipping: TestInfrastructure_GenerateResources did not succeed")
+	}
+
 	t.Log("Verifying credentials.yaml")
 
 	config := NewTestConfig()
@@ -139,6 +152,10 @@ func TestInfrastructure_VerifyCredentialsYAML(t *testing.T) {
 
 // TestInfrastructure_VerifyInfrastructureSecretsYAML verifies is.yaml exists and is valid
 func TestInfrastructure_VerifyInfrastructureSecretsYAML(t *testing.T) {
+	if !infrastructureGenerationSucceeded {
+		t.Skip("Skipping: TestInfrastructure_GenerateResources did not succeed")
+	}
+
 	t.Log("Verifying is.yaml (infrastructure secrets)")
 
 	config := NewTestConfig()
@@ -170,6 +187,10 @@ func TestInfrastructure_VerifyInfrastructureSecretsYAML(t *testing.T) {
 
 // TestInfrastructure_VerifyAROClusterYAML verifies aro.yaml exists and is valid
 func TestInfrastructure_VerifyAROClusterYAML(t *testing.T) {
+	if !infrastructureGenerationSucceeded {
+		t.Skip("Skipping: TestInfrastructure_GenerateResources did not succeed")
+	}
+
 	t.Log("Verifying aro.yaml (ARO cluster configuration)")
 
 	config := NewTestConfig()

